kubernetes-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/ApplyMojo.java

```diff
@@ -144,8 +144,8 @@ public class ApplyMojo extends AbstractJKubeMojo implements ManifestProvider {
     protected ApplyService applyService;
 
     @Override
-    protected boolean canExecute() {
-        return super.canExecute() && !skipApply;
+    protected boolean shouldSkip() {
+        return super.shouldSkip() && skipApply;
```

Comment:
This doesn't look correct. A goal/task can be skipped if either one of `jkube.skip` or `jkube.skip.goal` options is enabled. I think you should do this here:

```suggestion
        return super.shouldSkip() || skipApply;
```